opentelemetry-sdk/src/opentelemetry/sdk/metrics/view.py

```diff
@@ -0,0 +1,185 @@
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import logging
+import threading
+from collections import defaultdict
+from typing import Optional, Sequence, Tuple
+
+from opentelemetry.metrics import (
+    Counter,
+    InstrumentT,
+    SumObserver,
+    UpDownCounter,
+    UpDownSumObserver,
+    ValueObserver,
+    ValueRecorder,
+    ValueT,
+)
+from opentelemetry.sdk.metrics.export.aggregate import (
+    Aggregator,
+    LastValueAggregator,
+    MinMaxSumCountAggregator,
+    SumAggregator,
+    ValueObserverAggregator,
+)
+
+logger = logging.getLogger(__name__)
+
+
+class ViewData:
+    def __init__(self, labels: Tuple[Tuple[str, str]], aggregator: Aggregator):
+        self.labels = labels
+        self.aggregator = aggregator
+
+    def record(self, value: ValueT):
+        self.aggregator.update(value)
+
+    # Uniqueness is based on labels and aggregator type
+    def __hash__(self):
+        return hash((self.labels, self.aggregator.__class__))
+
+    def __eq__(self, other):
+        return (
+            self.labels == other.labels
+            and self.aggregator.__class__ == other.aggregator.__class__
+        )
+
+
+class ViewConfig:
+
+    UNGROUPED = 0
+    LABEL_KEYS = 1
+    DROP_ALL = 2
+
+
+class View:
+    def __init__(
+        self,
+        metric: InstrumentT,
+        aggregator: type,
+        aggregator_config: Optional[dict] = None,
+        label_keys: Optional[Sequence[str]] = None,
+        view_config: ViewConfig = ViewConfig.UNGROUPED,
+    ):
+        self.metric = metric
+        self.aggregator = aggregator
+        if aggregator_config is None:
+            aggregator_config = {}
+        self.aggregator_config = aggregator_config
+        if label_keys is None:
+            label_keys = []
+        self.label_keys = sorted(label_keys)
+        self.view_config = view_config
+        self.view_datas = set()
+
+    def get_view_data(self, labels):
+        """Find an existing ViewData for this set of labels. If that ViewData
+            does not exist, create a new one to represent the labels
+        """
+        active_labels = []
+        if self.view_config == ViewConfig.LABEL_KEYS:
+            # reduce the set of labels to only labels specified in label_keys
+            active_labels = {
+                (lk, lv) for lk, lv in labels if lk in set(self.label_keys)
+            }
+            active_labels = tuple(active_labels)
+        elif self.view_config == ViewConfig.UNGROUPED:
+            active_labels = labels
+
+        for view_data in self.view_datas:
+            if view_data.labels == active_labels:
+                return view_data
+        new_view_data = ViewData(
+            active_labels, self.aggregator(self.aggregator_config)
+        )
+        self.view_datas.add(new_view_data)
+        return new_view_data
+
+    # Uniqueness is based on metric, aggregator type, aggregator config, ordered label keys and ViewConfig
```

Comment:
@cnnradams please wrap comments at 80 chars. It'd be great if `black` did this for us, but alas.